universum/configuration_support.py

```diff
@@ -8,14 +8,133 @@
     "Step",
     "Configuration",
     "Variations",
+    "combine",
     "set_project_root",
     "get_project_root"
 ]
 
 
+# pylint: disable-msg=line-too-long
 class Step:
     """
-    Step is a collection of configurable project-specific data needed for Universum operations.
+    Step is a collection of configurable project-specific data entries needed for Universum functionality. Individual
+    build steps are collected in a :class:`Configuration` object, which supports arithmetic operations for
+    combination and chaining.
+
+    **Step keys**
+
+    Parameters supplied via named attributes during construction are annotated and can be type-checked for safety.
+    Clients may also provide custom parameters in kwargs - these parameters will be stored internally in `dict` and can
```

Comment:
I'd rather use term 'users', because it's also used everywhere else in Universum documentation.
Also, I'd prefer `kwargs` instead of kwargs.
Also, I'd say 'in a `dict` field/attribute' instead of simple 'in `dict`'.
Also, I'd replace 'provide' with 'define'.